packages/cmr-client/src/CMR.ts

```diff
@@ -131,6 +122,12 @@ export class CMR {
    * @returns {Promise.<string | undefined>} the token
    */
   async getToken(): Promise<string | undefined> {
+    if (this.oauthProvider === 'launchpad') {
+      return this.token;
+    }
+    if (!this.username) {
+      throw new Error('Username not specified for non-launchpad CMR client');
+    }
```

Comment:
is this a fix for a problem introduced in this PR? or is this just a bug fix while you're making changes?

Reply:
Sort-of.     The fix to the class typing above was directly related to changes in the recon report (adding typing)  and that change cascaded into needing to handle this case correctly here.